### DISCUSSION TITLE
Build issue/error uin Vercel - Nextjs, Graphql, and Datocms

### DISCUSSION THREAD

**Discussion by vishalshah6153** [vercel/next.js#44846] [Help]
Hi there, I'm managing multiple Nextjs + Datocms projects. And recently I started having this issue with the vercel build. For most of my projects, the build is failing with the same error. Even after reverting to the latest changes, it gives the same error. It isn't building for the same code that was previously built successfully. The error is:

![image](https://user-images.githubusercontent.com/89897183/212267158-dac88a66-ddf9-4b91-95cc-db3d4529d765.png)

Usually, this was the error I get when there is some syntax error in the graphql query but I've confirmed that all the queries are correct.
I've also integrated the web preview plugin that Dato provides. So when I open the preview with the URL as: ``${baseUrl}${url}?preview=exit``   it works fine in the local environment but not in the staging. After analyzing the network tab it seems that the /api/graphql endpoint gives the same error that's occurring in the vercel build:  

![image](https://user-images.githubusercontent.com/89897183/212260856-94a61214-4329-42c2-8803-b8d009d702a4.png)

So my assumption is that both the issues are the same or somehow connected. I had put some console logs in the graphql file to see what's the error but the logs don't appear in the console at all as if the endpoint isn't getting called. Although in the network tab, the same endpoint shows a 500 error. So I'm very confused and can't get any lead on the issue. Can someone please help?

**Comment by evangelinepapanicola** [comment]
@vishalshah6153 Have you tried checking the function logs on your Vercel project? If you open the errors tab there, and then refresh your /api/graphql endpoint it may display an error to help point you in the right direction:
![image](https://user-images.githubusercontent.com/33719860/212690154-a977837b-e6eb-4188-911e-33673d036e6a.png)

**Reply by vishalshah6153** [reply]
@evangelinepapanicola  Yes, I did. And the logs show successful function calls as well.
